Changes

```diff
@@ -18,6 +18,10 @@ Working version
   `Thread.sigmask` using `Unix.sigprocmask`.
   (Xavier Leroy, review by Antonin Décimo, Gabriel Scherer, Miod Vallat)
 
+* #13435: On Windows, use `GetTempPath2`/`GetTempPath` for
+  `Filename.get_temp_dir_name`.
```

Comment:
Thinking about what would be useful to the reader of the Changes entry, I would tweak the text bit:
```suggestion
* #13435: On Windows, use native Windows API to retrieve the location of the
  temporary directory returned by `Filename.get_temp_dir_name`. Consequently,
  different results may be returned in certain situations. See the docstring of the
  function in  the `Filename` module for more details.
```

Reply:
Actually, I'd go a step further - the reason for this change is not that Rust, Go and everyone else do it - it's that it switches to a more secure version of the API, reducing the attack surface on an OCaml process which is being used in a secure context (cf. #7557 for why this kind of thing can unexpectedly matter and even [CVE-2024-38081](https://www.cve.org/CVERecord?id=CVE-2024-38081) for a recent example!).
```suggestion
* #13435: On Windows, use system calls for `Filename.get_temp_dir_name` instead
  of directly reading the environment, which in particular improves the
  security of OCaml processes running in the SYSTEM security context by
  mitigating privileged file operation attacks. For all other processes running
  with the default environment (where TEMP is set), there is no discernible
  change.
```